src/evo/deterministicmns.cpp

```diff
@@ -1327,7 +1327,8 @@ static std::optional<ProTx> GetValidatedPayload(const CTransaction& tx, gsl::not
         return std::nullopt;
     }
     const bool is_basic_scheme_active{DeploymentActiveAfter(pindexPrev, Params().GetConsensus(), Consensus::DEPLOYMENT_V19)};
-    if (!opt_ptx->IsTriviallyValid(is_basic_scheme_active, state)) {
+    const bool is_extended_addr{DeploymentActiveAfter(pindexPrev, Params().GetConsensus(), Consensus::DEPLOYMENT_V23)};
```

Comment:
Consider implementing new helper which converts CBlockIndex* to `ProTxVersion`.

Something like
```cpp
inline ProTxVersion ProtxVersionAt(const CBlockIndex& index, const Consensus::Params& params)
{
 if (DeploymentActiveAfter(pindexPrev, Params().GetConsensus(), Consensus::DEPLOYMENT_V23)) return ProTxVersion::Ext;
 return (v19...) ?   ProTxVersion::Basic :  ProTxVersion::Legacy;  }
```

Passing 2 flags deep inside `IsTriviallyValid` and `GetMaxVersion` does look too good